zio-http/shared/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -402,6 +402,20 @@ object JsonSchema {
           ref,
           leftSchema.children ++ rightSchema.children,
         )
+      case Schema.Fallback(left, right, fullDecode, _)                                       =>
+        val leftSchema  = fromZSchemaMulti(left, refType)
+        val rightSchema = fromZSchemaMulti(right, refType)
+        val innerSchema =
+          if (fullDecode)
+            AnyOfSchema(Chunk(leftSchema.root, rightSchema.root))
+          else
+            OneOfSchema(Chunk(leftSchema.root, rightSchema.root))
```

Comment:
I wonder if I understood `Schema.Fallback.fullDecode` correctly here.

Reply:
@987Nabil Thanks for clarifying. I assumed `Schema.Fallback` was equivalent to [`Schema.Either`](https://github.com/zio/zio-http/pull/2730/files/c2a49d6cf3afc22582219242b4f22a113c02b626#diff-1dc286225d0eee11fb5608307e6eef3097d66c5b49426a2d76fc78ba0f05536aR401) when `fullDecode` is off, using `oneOf`.

Just double-checking: is `anyOf(left, right)` sufficient, or should we consider adding a variant for `fullDecode`, like `oneOf(left, right, allOf(left, right))`?

Reply:
That's a good question. Thinking about it the idea of `oneOf(left, right, allOf(left, right))` is more precise. And I don't see a reason not to do it.